src/sqlfluff/rules/L003.py

```diff
@@ -882,15 +882,11 @@ def segment_info(idx: int) -> Tuple[str, Optional[str]]:
         if not cls._single_placeholder_line(current_line):
             return False
         for idx in range(1, len(current_line)):
-            if (
-                segment_info(idx - 1)
-                in (
-                    ("placeholder", "block_start"),
-                    ("placeholder", "compound"),
-                    ("placeholder", "block_mid"),
-                )
-                and segment_info(idx) == ("indent", None)
-            ):
+            if segment_info(idx - 1) in (
+                ("placeholder", "block_start"),
+                ("placeholder", "compound"),
+                ("placeholder", "block_mid"),
+            ) and segment_info(idx) == ("indent", None):
```

Comment:
OK this is an interesting one. Black 22.1.0 prefers this without the outer brackets.
Versions less than that prefer it with.

Unfortunately dbt needs an. older version of black so we can't pin it to 22.1.0 to ensure this doesn't revert in future.

We could pin to `<=21.12b0` but that feels kind of wrong and limiting for the future.

So currently it all works, but if a user has an older version of black (e.g. because they use dbt) then they might try to revert this change. That will fail in our CI and may lead to confusion.

But can't think of a better solution.

Reply:
If we want, we could install Black in a separate Python environment from dbt. At my last job, I eventually identified and separated a number of tools in this way -- they didn't have to be in the same Python environment because they didn't `import` or otherwise call other packages, they just did "system" things such as reading from files, writing to the terminal etc. This included most of the linting kind of tools, but not the ones that ran automated tests (e.g. pytest).

Reply:
We already do for linting (tox sets up it's own env). The problem is dbt installs it. So if using that env it will use that black. And some people (including you!) don't use tox.

Then again dbt tests seem to be freezing anyway. So might might end up freezing black anyway!

Reply:
OK I'm sticking with pinning the older version of black for now. Only thing that seems to work.

I'll pin flake8-black for now too for when we unpin black.